### ISSUE TITLE
Question about Primary Key

### ISSUE THREAD

**Issue by privateOmega** [knorm/knorm#244]
> Whether or not the field is the primary field. NOTE: a model can only have one primary field, any subsequent primary fields will overwrite the existing one (also in child models).

@joelmukuthu I stumbled upon this while going through the documentation. Do you have a plan to remove this limitation anytime in the future? As it would prevent  a person to  create **Composite Primary Key** and I think this should be addressed ASAP.

**Comment by joelmukuthu** [comment]
Yeah, composite primary keys are not supported at the moment and no, I didn't have plans to address this right away. It would be easy enough to add support for more than one primary field but there are some questions on implementation and usage that I haven't answered yet:

1. Should it be implemented by allowing `primary: true` on more than one field, or `Model.primary = ['field1', 'field2']`, which more closely resembles SQL? If we go with the second one, then we remove the first to avoid confusion.
2. If a field is added in a composite key, can it still be configured as a unique field?
3. In @knorm/relations, rows are indexed by the primary field when parsing database results. How would those be handled?
4. In @knorm/postgres, updating multiple rows in one query is supported, via `UPDATE ... FROM ...`, and the primary field is used in the `WHERE` clauses. How do we handle that with composite keys?
5. `Query.prototype.save` and `Model.prototype.save` use the primary field to figure out whether or not they should do an update or insert. That also a strategy for handling composite keys.
6. There might be others..

It'll need some careful consideration and spec'ing which I haven't prioritized at the moment. Reason being, one can still use knorm to send the right queries when working with composite fields. Once these questions have answers the implementation should be rather straightforward, and you are more than welcome to help in figuring them out :)
